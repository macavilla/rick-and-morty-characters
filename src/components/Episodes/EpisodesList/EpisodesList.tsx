import { EpisodeType } from "@/types/episode";

interface EpisodesListProps {
  title: string;
  episodes: EpisodeType[] | null;
}

const EpisodesList = ({ title, episodes }: EpisodesListProps) => {
  return (
    <div>
      <h4 className="text-2xl">{title} </h4>

      {episodes && (
        <ul>
          {episodes.map((episode) => (
            <li className="my-3 list-disc list-inside" key={episode.id}>
              {episode.episode} - {episode.name} - {episode.air_date}
            </li>
          ))}
        </ul>
      )}
    </div>
  );
};

export default EpisodesList;
