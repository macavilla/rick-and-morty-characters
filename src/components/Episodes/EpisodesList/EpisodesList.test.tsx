import { screen, render } from "@testing-library/react";
import { mockEpisodesListProps } from "./__mockData__/EpisodesList.mock";

import EpisodesList from "../EpisodesList/EpisodesList";

describe("EpisodesList", () => {
  it("renders CharacterCard and matches snapshot", () => {
    const { container } = render(<EpisodesList {...mockEpisodesListProps} />);
    expect(container).toMatchSnapshot();
  });

  it("should render three EpisodesListList with shared episodes data", () => {
    render(<EpisodesList {...mockEpisodesListProps} />);
  });
});
