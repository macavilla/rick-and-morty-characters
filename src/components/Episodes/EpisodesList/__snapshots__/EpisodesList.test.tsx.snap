// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`EpisodesList renders CharacterCard and matches snapshot 1`] = `
<div>
  <div
    class="px-4"
  >
    <h4
      class="text-2xl text-center"
    >
      Test Episodes List
       
    </h4>
    <ul>
      <li
        class="my-3 list-disc list-inside"
      >
        S01E01
         - 
        Pilot
         - 
        December 2, 2013
      </li>
      <li
        class="my-3 list-disc list-inside"
      >
        S01E02
         - 
        Lawnmower Dog
         - 
        December 9, 2013
      </li>
      <li
        class="my-3 list-disc list-inside"
      >
        S01E03
         - 
        Anatomy Park
         - 
        December 16, 2013
      </li>
    </ul>
  </div>
</div>
`;
